src/lepton/recoder.cc

```diff
@@ -850,20 +844,20 @@ bool recode_baseline_jpeg(bounded_iostream*str_out,
     if (!str_out->has_reached_bound() ) {
         str_out->write( hdrdata + byte_position, hdrs - byte_position );
     }
-    if (ujgversion != 1) {
-        for (size_t i = 0; i < NUM_THREADS; ++i) {
-            delete huffws[i];
+    
```

Comment:
Why was this check removed?

Reply:
Because it leaked memory. I should ask author why this check was added in the first time.

Reply:
I emailed author and moved this commit to the next PR in order to get his answer first.